README.md

```diff
@@ -0,0 +1,400 @@
+The author of this RFC feels like addressing the `Pin` soundness in the definitions of the traits is wrong, as in almost all cases where are a user implements one of these traits `Pin` will be irrelevant to them. And at its core, unsizing is not coupled with `Pin` whatsoever.
+It would make much more sense to fix the unsound `CoerceUnsized` implementation that `Pin` provides.
+That is given the current implementation of (with the new definition of the trait):
+```rs
+
+// Copied from core library docs:
+// Note: this means that any impl of `CoerceUnsized` that allows coercing from
+// a type that impls `Deref<Target=impl !Unpin>` to a type that impls
+// `Deref<Target=Unpin>` is unsound. Any such impl would probably be unsound
+// for other reasons, though, so we just need to take care not to allow such
+// impls to land in std.
+impl<P, U> CoerceUnsized<Pin<U>> for Pin<P>
+where
+    P: CoerceUnsized<U>,
+    // U: core::ops::Deref, this bound is accidentally missing upstream, hence we can't use `Pin::new_unchecked` in the implementation
+{
+    fn coerce_unsized(self) -> Pin<U> {
+        Pin {
+            pointer: self.pointer.coerce_unsized(),
+        }
+    }
+}
+```
+
+We should rather strife to have the following 2 implementations:
+```rs
+// Permit going from Pin<impl Unpin> to Pin<impl Unpin>
+impl<P, U> CoerceUnsized<Pin<U>> for Pin<P>
+where
+    P: CoerceUnsized<U>,
+    P: Deref<Target: Unpin>,
+    U: Deref<Target: Unpin>,
+{
+    fn coerce_unsized(self) -> Pin<U> {
+        Pin::new(self.pointer.coerce_unsized())
+    }
+}
+// Permit going from Pin<impl Pin> to Pin<impl Pin>
+impl<P, U> CoerceUnsized<Pin<U>> for Pin<P>
+where
+    P: CoerceUnsized<U>,
+    P: core::ops::Deref<Target: !Unpin>,
+    U: core::ops::Deref<Target: !Unpin>,
+{
+    fn coerce_unsized(self) -> Pin<U> {
+        // SAFETY: The new unpin Pin is derived from another unpin Pin, so we the pinned contract is kept up
+        unsafe { Pin::new_unchecked(self.pointer.coerce_unsized()) }
+    }
+}
+```
+
+
+While this is a breaking change, it should be in line with being a soundness fix.
+Unfortunately, these kind of impl requires negative bounds and negative reasoning which is its own can of worms and therefor likely not to happen, see https://github.com/rust-lang/rust/issues/42721.
+Though maybe allowing them for auto traits alone could work out fine, given those types of traits are already rather special.
+
+## Custom Reborrows
+
+See the following issue for context: https://github.com/rust-lang/rfcs/issues/1403
+
+In the linked issue the idea was to generalize `CoerceUnsized` as a general `Coerce` trait.
+
+The design proposed here would still enable this, although it raises some questions.
+- For one, reborrows should be guaranteed to be no-ops, as all that should change in "reborrow coercions" are the corresponding lifetimes, yet this RFC exposes a function that will be run on coercion.
```

Comment:
Does "this RFC" refer to this RFC or to RFC 1403?

Reply:
This one here. Note that reborros has no rfc, just an "idea issue"